### ISSUE TITLE
Completion for `std::ptr::NonNull::new_unchecked` is weird

### ISSUE THREAD

**Issue by kojole** [racer-rust/racer#877]
```
$ racer complete std::ptr::NonNull::
MATCH dangling,2688,11,/Users/kojole/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/../libcore/ptr.rs,Function,pub fn dangling() -> Self
MATCH unsafe fn new_unchecked(ptr,2704,14,/Users/kojole/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/../libcore/ptr.rs,Const,pub const unsafe fn new_unchecked(ptr: *mut T) -> Self {
MATCH new,2710,11,/Users/kojole/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/../libcore/ptr.rs,Function,pub fn new(ptr: *mut T) -> Option<Self>
MATCH from,2821,7,/Users/kojole/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/../libcore/ptr.rs,Function,fn from(unique: Unique<T>) -> Self
MATCH from,2829,7,/Users/kojole/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/../libcore/ptr.rs,Function,fn from(reference: &'a mut T) -> Self
MATCH from,2837,7,/Users/kojole/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src/libstd/../libcore/ptr.rs,Function,fn from(reference: &'a T) -> Self
```

`std::ptr::NonNull::new_unchecked` is `const unsafe fn`, and I suspect racer's matcher doesn't handle `const unsafe fn`s correctly ([src/racer/matchers.rs#L189](https://github.com/racer-rust/racer/blob/523ffb3fc750287b93daf71ed4c48429df4fde35/src/racer/matchers.rs#L189)).

### Versions
- rustc 1.27.0 (3eda71b00 2018-06-19)
- racer 2.0.14

**Comment by kngwyu** [comment]
Thanks for reporting, and sorry for poor experience :fearful: 
Racer's keyword mathing is actually not good.

**Comment by kngwyu** [comment]
closed via #879
